Maskbook/Scene/App/FIleService/Controller/FileServiceDetailViewController.swift

```diff
@@ -23,114 +18,41 @@ final class FileServiceDetailViewController: BaseViewController {
         super.viewDidLoad()
         title = fileServiceItem.fileName
 
+        let viewModel = FileServiceDetailViewModel(item: fileServiceItem)
         FileServiceDetailView(
-            item: fileServiceItem,
-            action: { [weak self] action in
-                guard let self = self else { return }
-                switch action {
-                case .share: self.share()
-                case .download: self.download()
-                }
-            }
+            viewModel: viewModel
         )
         .asContent(in: self)
     }
-
-    private func share() {
-        guard let shareItem = FileServiceUploadResult.from(fileServiceItem) else {
-            log.info("FileServiceUploadResult get a nil value")
-            return
-        }
-        shareViewModel.postFileService(fileServiceResult: shareItem)
-    }
-
-    private func download() {
```

Comment:
I insist that we  keep the navigation logic in `ViewController` unless we can't pass out the navigation event.

Reply:
OK, I will restore your part of that